source/common/upstream/ring_hash_lb.cc

```diff
@@ -16,8 +16,13 @@ RingHashLoadBalancer::RingHashLoadBalancer(
     const Optional<envoy::api::v2::Cluster::RingHashLbConfig>& config)
     : host_set_(*priority_set.hostSetsPerPriority()[0]), stats_(stats), runtime_(runtime),
       random_(random), config_(config) {
-  host_set_.addMemberUpdateCb([this](uint32_t, const std::vector<HostSharedPtr>&,
-                                     const std::vector<HostSharedPtr>&) -> void { refresh(); });
+  priority_set.addMemberUpdateCb([this](uint32_t priority, const std::vector<HostSharedPtr>&,
+                                        const std::vector<HostSharedPtr>&) -> void {
+    if (priority != 0) {
+      return; // TODO(alyssawilk) make ring hash LB priority-aware.
```

Comment:
Can we just block this inside EdsClusterImpl? The cluster knows it's LB type and we could check it there and fail the config. I'm not sure when/if we will make ring hash priority aware. Then we could switch this to assert.

In general, I'm wondering if we could do all the config blocking inside EdsClusterImpl. We could even add a ClusterManager method to get the "local cluster" so that an EDS cluster could tell if it is the local cluster and then fail if that cluster has more than 1 priority.

Reply:
FWIW ring hash code is written and tested.  It depends on this one so I can't send it out yet. :-)

I'll take a look at what it'd take to do config validation here.  I'd wanted to do an end to end failed update in the load balancing test and I think that's going to be complicated.

Reply:
If you are going to do it soon it's not a big deal, was just thinking in general it would be better to block config in EDS updates vs. silently ignore here.

Reply:
Sure thing, done.